src/treelearner/cuda/cuda_data_partition.cu

```diff
@@ -1045,11 +1045,11 @@ __global__ void AddPredictionToScoreKernel(
       const data_size_t global_data_index = data_indices_in_leaf[local_data_index];
       const int leaf_index = cuda_data_index_to_leaf_index[global_data_index];
       const double leaf_prediction_value = leaf_value[leaf_index];
-      cuda_scores[local_data_index] = leaf_prediction_value;
+      cuda_scores[global_data_index] += leaf_prediction_value;
```

Comment:
this is bug ?

Reply:
No. When USE_BAGGING is false, `data_indices_in_leaf` will be all the data indices of training data. So `local_data_index` is exactly `global_data_index`.